src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -25,6 +26,31 @@ public HostString(string value)
         }
 
         /// <summary>
+        /// Creates a new HostString without modification.
+        /// </summary>
+        /// <param name="value">The value should be Unicode rather than punycode. IPv6 addresses must use square braces.</param>
```

Comment:
> `value`

Really? Looks more like the `domain` from how it's handled.

> must use square braces

No "must"; the constructor adds missing square braces.